### DISCUSSION TITLE
Multi-level graph embeddings

### DISCUSSION THREAD

**Discussion by alzaia** [pyg-team/pytorch_geometric#5971] [Q&A] [answered]
Hello guys, I have a use case where I want to see if combining 2 different levels of my graphs could help for my task. I have my whole graph of nodes/edges where I'm doing node classification. These nodes are also part of subgraph ids (which I do not use for now). What I wanted is, in parallel of my node embeddings operation, to do a subgraph embedding operation that would learn subgraph features that could then be used for my node classification task. For example: node `i` has its node features `Fi`, but is also part of subgraph `m` that has subgraph features `Fm`. Are there any layers/models in PyG that would be worth checking out? Or just relevant similar examples to my case? Thanks a lot!

**Comment by rusty1s** [comment, marked as answer]
This sounds super similar to one of my research topics, see [here](https://arxiv.org/abs/2006.12179). Happy to help with any implementation questions.

**Reply by alzaia** [reply]
Thanks @rusty1s for the always fast and helpful replies! I checked the links, do you have this implementation already in PyG, or at least something very similar to that concept of hierarchical embeddings in PyG? I wanted to do a quick first experiment first before going full into it.

**Reply by rusty1s** [reply]
There is a PyG implementation [here](https://github.com/rusty1s/himp-gnn), but I haven't run it in a long time :(
